### PR TITLE
CI Improvements

### PR DESCRIPTION
**Pull request by greens**
Differentiates between failures and errors when running tests Ensures a clean working directory in Jenkins when deploying drivers Delete driver channel invites when the PR that created it is closed Adds a small sleep after zipping the driver package to ensure correct execution of deploy script

### CODE REVIEW

**Review by sjl2178 — APPROVED** [review]

**Review by ctowns — APPROVED** [review]

File: tools/deploy.py [hunk 1]
```diff
@@ -80,7 +80,8 @@
   if package_key == "bose" and BOSE_APPKEY:
     # write the app key into a app_key.lua (overwrite if exists already)
     subprocess.run(["touch -a ./src/app_key.lua && echo \'return \"" + BOSE_APPKEY +  "\"\n\' > ./src/app_key.lua"], cwd=driver, shell=True, capture_output=True)
-  subprocess.run(["zip -r ../edge.zip config.yml fingerprints.yml $(find profiles -name \"*.y*ml\") $(find . -name \"*.lua\") -x \"*test*\""], cwd=driver, shell=True,  capture_output=True)
+  subprocess.run(["zip -r ../edge.zip config.yml fingerprints.yml $(find profiles -name \"*.y*ml\") $(find . -name \"*.lua\") -x \"*test*\""], cwd=driver, shell=True, capture_output=True)
```
Comment (dljsjr) [inline tools/deploy.py:83]: Do you have magical white-space detecting glasses or did your Python formatter find this for you?

Comment (greens) [reply tools/deploy.py:83]: I just saw it.

File: tools/deploy.py [hunk 2]
```diff
@@ -80,7 +80,8 @@
...
+  time.sleep(.5)
```
[the remark this answers is not in the document]
Comment (greens) [reply tools/deploy.py:84]: We occasionally get deploy errors on Jenkins and my best guess (and the best guess of RelEng) is that there's a bit of a race condition here. `subprocess` should block until execution completes, but my guess is there's a tiny bit of filesystem lag.

File: .github/workflows/delete-channel.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          comment-author: 'github-actions[bot]'
+          issue-number: ${{ github.event.number }}
+      - name: post comment
+        if: ${{ steps.fc.outputs.comment-id == 0 }}
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          body: |
+            Channel deleted.
+          edit-mode: replace
+          issue-number: ${{ github.event.number }}
```
[the remark this answers is not in the document]
Comment (greens) [reply .github/workflows/delete-channel.yml:49]: there are a lot of things in the marketplace. they can be pretty hit-or-miss

**Pull request merged**